apps/system/lockscreen/js/lockscreen_inputpad.js

```diff
@@ -92,20 +91,19 @@
   LockScreenInputpad.prototype.handleEvent = function(evt) {
     switch (evt.type) {
       case 'lockscreen-notify-passcode-validationfailed':
-        return this.decorateErrorPasscodeUI()
-          .then(() => {
-            this.removeErrorPasscodeUI();
-            this.updatePassCodeUI();
-          });
+        this.states.passCodeErrorTimeoutPending = true;
+        this.updatePassCodeUI();
+        break;
+      case 'lockscreen-notify-passcode-validationreset':
```

Comment:
Although I can understand the possible reason why we need to have two events for the same effects, I think it is better to leave some comments about that. Since according the event flow it's quite strange without knowing the current status of LockScreen:

  validationfailed ----> validationreset  === validationsuccess

So I think we should add few lines here, to explain when the reset would be fired. That is, after the failure and we need to reset the UI.